src/frontend/js/project-service.js

```diff
@@ -967,4 +1150,109 @@ angular.module('seashell-projects', ['seashell-websocket', 'marmoset-bindings'])
       self.open = function(name, lock) {
         return (new SeashellProject(name)).init(lock);
       };
+
+
+      /**
+       * Sync an entire project with the offline store. 
+       * Only files that already exist online will be synced, 
+       * so this function is more useful for dumping into the offline store.
+       * @param {String} project name
+       * @returns {Angular.$q -> true}
+       *  Angular deferred that resolves to to true when the project is stored.
+       */
+      self.syncProject = function(name) {
```

Comment:
We shouldn't have complicated syncing logic here, as we're going to need to sync from within the websocket service (offline syncs must run when reconnected).